torch/fx/experimental/GraphManipulation.py

```diff
@@ -61,18 +69,157 @@ def get_size_of_node(fx_module: GraphModule, node: Node) -> size_bytes:
+    weight["shape"] = serialize_shape(tensor.shape)
+    return weight
+
+
+def serialize_module(fx_module: GraphModule, weights: Dict, name_prefix="") -> Dict:
+    """Recursively Serializes a graph module (fx_module) to a dictionary which is later exported to JSON.
+    It also adds all weights the the provided weights dictionary by qualified_name.
+    Dictionary Schema:
+    MODULE
+    {
+        modules: {module_name: MODULE],
+        nodes: [NODE],
+        weights {qualified_name: WEIGHT},
+    }
+    NODE
+    {
+        shape: [],
+        dtype: dtype,
+        target: target,
+        op_code: op_code,
+        name: name,
+        args: [],
+        kwargs: {}
+    }
+    WEIGHT
+    {
+        dtype: dtype,
+        is_quantized: bool,
+        shape: [],
+        quantization_info: QUANTIZATION
+    }
+    QUANTIZATION
+    {
+        qsheme: sqcheme,
+        q_scale: float,
+        q_zero_point: float,
+        q_per_channel_scales, [],
+        q_per_channel_zero_points: [],
+        q_per_channel_axis, int
+    }
+    """
+    serialized_dict = {}
+    serialized_dict["modules"] = {}
+    serialized_dict["weights"] = {}
+    serialized_dict["nodes"] = []
+    parameters = fx_module.named_parameters()
+    for name, p in parameters:
+        if isinstance(p, torch.Tensor):
+            weight = serialize_weight(p)
+            prefix = f"{name_prefix}." if name_prefix else ""
+            serialized_dict["weights"][prefix + name] = weight
+            weights[prefix + name] = p
+    for node in fx_module.graph.nodes:
+        node_rep = {}
+        # Get shape/type info, currently not needed for call_module.
+        if node.op != "call_module":
+            shape = getattr(node, "shape", None)
+            if shape:
+                node_rep["shape"] = serialize_shape(shape)
+            else:
+                raise RuntimeError(
+                    "Node has no shape attr, this is likely because shape propigation has not been run on this Graph."
+                )
+            dtype = getattr(node, "dtype", None)
+            if dtype:
+                node_rep["dtype"] = str(dtype)
+            else:
+                raise RuntimeError(
+                    "Node has no dtype attr, this is likely because shape propigation has not been run on this Graph."
+                )
+
+        # Recurse down into any submodules we are calling.
+        if node.op == "call_module":
+            submodules = dict(fx_module.named_modules())
+            if isinstance(submodules[node.target], torch.fx.GraphModule):
+                serialized_module = serialize_module(
+                    getattr(fx_module, node.target), weights, node.target
+                )
+                serialized_dict["modules"][node.target] = serialized_module
+
+        if node.op == "call_function":
+            node_rep["target"] = get_qualified_name(node.target)
+        else:
+            node_rep["target"] = str(node.target)
```

Comment:
It's because only call function can have python builtins as node.target.

Reply:
Do we need to special case this, since it seems like `get_qualified_name` handles this? I.e. it contains:

```python
    if getattr(builtins, func.__name__, None) is func:
        return func.__name__	        return func.__name__
```

Reply:
I think it's the flip, node.target is a string for other nodes, so get_qualified_name won't handle them. We only need the special handling for call_function node.target